### PR TITLE
Add support for dim=3 in ttnn.argmax

### PR DESCRIPTION
**Pull request by sraizada-tt**
### Ticket
#12386

### Problem description
ttnn.argmax only supported dim=None 

### What's changed
Added support for dim=3 in ttnn.argmax

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/10937570798
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_argmax.py [hunk 1]
```diff
@@ -49,15 +49,15 @@ def run_argmax_tests(
...
+    tt_result = tt_result.squeeze(2)
     assert len(tt_result.shape) == len(ref_value.shape)
     assert tt_result.shape == ref_value.shape
     assert_with_pcc(ref_value, tt_result, 0.99)
 
 
 test_sweep_args = [
     (
-        [(1, 3, 19, 98)],
+        [(1, 1, 19, 98)],
```
Comment (sjameelTT) [inline tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_argmax.py:60]: Any reason we're getting rid of the old dims? Is the squeeze necessary/should it be two separate tests? IIRC argmax was brought up by Brian for Moreh and those dims may be important for them.

File: tests/ttnn/profiling/ops_for_profiling.py [hunk 2]
```diff
@@ -2160,13 +2160,13 @@ def assign_unary(x):
...
     },
-    {
-        "op": argmax_dim_2,
-        "name": "ttnn.argmax_dim_2",
-        "shape_func": argmax_shape_func,
-        "layout": "ROW_MAJOR",
-        "num_repeats": 2,
-    },
+    # {
+    #     "op": argmax_dim_2,
```
Comment (sjameelTT) [inline tests/ttnn/profiling/ops_for_profiling.py:2164]: Same here, any reason this config is commented out?

**Review by sjameelTT — APPROVED** [review]
Just some questions

File: tests/ttnn/profiling/ops_for_profiling.py (hunk 2 above)
Comment (sraizada-tt) [reply tests/ttnn/profiling/ops_for_profiling.py:2164]: this was never functional -- dim was defaulted to None

File: tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_argmax.py (hunk 1 above)
Comment (sraizada-tt) [reply tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_argmax.py:60]: this test wasn't working, hence under non_working_unit_tests

File: tests/ttnn/profiling/ops_for_profiling.py (hunk 2 above)
Comment (yan-zaretskiy) [reply tests/ttnn/profiling/ops_for_profiling.py:2164]: Let's not comment out code, just remove it if it's not needed.

File: tests/ttnn/unit_tests/test_multi_device_async.py [hunk 3]
```diff
@@ -243,10 +243,14 @@ def test_multi_device_data_parallel_op_chain(pcie_mesh_device, program_cache, in
         pcie_mesh_device.get_device(device).enable_async(False)
 
 
-@pytest.mark.parametrize("layout", [ttnn.TILE_LAYOUT, ttnn.ROW_MAJOR_LAYOUT])
+@pytest.mark.parametrize(
+    "layout",
+    [
+        ttnn.ROW_MAJOR_LAYOUT,
```
Comment (yan-zaretskiy) [inline tests/ttnn/unit_tests/test_multi_device_async.py:249]: Why do we remove `TILE_LAYOUT` here?
Edit: I see that it only supports row-major below.

File: ttnn/cpp/ttnn/operations/reduction/argmax/device/argmax_op.cpp [hunk 4]
```diff
@@ -76,8 +73,12 @@ operation::ProgramWithCallbacks ArgMax::create_program(
...
     const auto &input_tensor = input_tensors.at(0);
     const auto &output_tensor = output_tensors.at(0);
-
-    return detail::argmax_multi_core(input_tensor, output_tensor, this->dim);
+    if (this->dim.has_value()) {
+        const uint32_t input_rank = input_tensor.get_legacy_shape().rank();
+        const uint32_t normalized_dim = dim.value() < 0 ? dim.value() + input_rank : dim.value();
+        return detail::argmax_single_core(input_tensor, output_tensor, normalized_dim);
+    }
+    return detail::argmax_single_core(input_tensor, output_tensor, this->dim);
```
Comment (yan-zaretskiy) [inline ttnn/cpp/ttnn/operations/reduction/argmax/device/argmax_op.cpp:81]: Consider
```
uint32_t normalized_dim = dim.has_value() 
    ? *dim + input_tensor.get_legacy_shape().rank() * (*dim < 0)
    : dim;
return detail::argmax_single_core(input_tensor, output_tensor, normalized_dim);
```

**Review by yan-zaretskiy — APPROVED** [review]

**Pull request merged**